bingo/bingo-elastic/python/bingo_elastic/model/record.py

```diff
@@ -29,37 +30,50 @@ def __set__(self, instance: IndigoRecord, value: Dict):
 
 class WithIndigoObject:
     def __set__(self, instance: IndigoRecord, value: IndigoObject) -> None:
+        value.aromatize()
+
         fingerprints = (
             "sim",
             "sub",
         )
+
         for f_print in fingerprints:
             try:
                 setattr(instance, f"{f_print}_fingerprint", [])
                 setattr(instance, f"{f_print}_fingerprint_len", 0)
-                fp_ = [
-                    int(feature)
-                    for feature in value.fingerprint(f_print)
-                    .oneBitsList()
-                    .split(" ")
-                ]
-                setattr(instance, f"{f_print}_fingerprint", fp_)
-                setattr(instance, f"{f_print}_fingerprint_len", len(fp_))
+
+                fp_list = value.fingerprint(f_print).oneBitsList()
+                if fp_list:
+                    fp_ = [int(feature) for feature in fp_list.split(" ")]
+                    setattr(instance, f"{f_print}_fingerprint", fp_)
+                    setattr(instance, f"{f_print}_fingerprint_len", len(fp_))
```

Comment:
@AATDev21 @mkviatkovskii so if no fingerprint is created, we don’t set up some attributes. Is it happening often? Shouldn’t we throw some error that for this molecule we couldn’t generate fingerprint?

Reply:
The only reason not to have fingerprints is to be an empty molecule, right? I can't imagine any situation when we would need to have something like it in index, so probably we actually should. But what do you think we should do in this case with the empty molecule test?